extension/public/reddit.js

```diff
@@ -5,95 +5,117 @@ var Reddit = {
+          }else if (description.search(blockedWordsData[i]) != -1){
+            return true;
+          }
+      }
+      return false;
   },
 
-  setupPostUI: function () {
-    // adds hide button to each Reddit post
-    posts = document.querySelectorAll(this.linkSelector);
-    console.log(posts);
-    posts.forEach((post) => {
-      post.innerHTML += this.hideLinkHTML();
-      post
-        .querySelector(".reddit-action.hide")
-        .addEventListener("click", (event) => {
-          this.removeLink(event);
-        });
-    });
-    console.log("setup post UI");
-  },
+  //extract text from image using tesseract
+  //returns text
+  async ocr (url) {
+
+      const { createWorker } = Tesseract;
+      const worker = createWorker ({
+      logger: m => console.log(m)
+      });
+
+      await worker.load();
+      await worker.loadLanguage('eng');
+      await worker.initialize('eng');
+      const { data: { text } } = await worker.recognize(url);
+      await worker.terminate();
+      return text;
 
-  // calls removeLinkFromDOM to remove a specific post link from the DOM
-  removeLink: function (event) {
-    var button = event.currentTarget;
-    var post = button.parentNode.parentNode;
-    var link = post.getAttribute("id");
-
-    this.getDOMTitle(post);
-    this.getDOMLink(post);
-    try {
-      this.getDOMImageLink(post);
-    } catch (e) {
-      console.log("Not an image" + e);
-    }
-    // this.blurImage(post);
-    this.removeLinkFromDOM(link);
   },
 
-  // called removeLinkFromDOM to remove all loaded posts on the page
-  // can use this method to call our image processing on all posts automatically
-  removeLinkAll: function () {
-    var posts = document.querySelectorAll(this.linkSelector);
-    // console.log("Found " + posts.length + " Visible posts");
-    // console.log(posts);
-    posts.forEach((post) => {
-      post.style.display = "none";
-      post.querySelector(".reddit-action").style.display = "none";
-
-    });
+  //blurs the image in a post when blocked keyword is found
+  blurImage: function () {
+      var posts = document.querySelectorAll(this.linkSelector);
+
+      posts.forEach((post) => {
+      var link = post.getAttribute("id");
+      //link(id) of promotions in reddit exceed 50 characters
+      const linkMaxLength = 50;
+      if(link.length < linkMaxLength){
+
+          //checking if current post contains blocked keyword
+          //promise has to be processed to retrieve result data
+          const promise = this.filterText(post,function () {console.log("ocr done");})
+          .then(function(isFound){
+          //isFound = result of promise
+          //true if found
+          console.log(isFound); // debug
+          if(isFound) {
+              console.log(document.getElementsByClassName(link));
+              document.getElementById(link).style.filter = "blur(5Px)";
+              //removes link if no images found.
+              //this.removeLinkFromDOM(link);
```

Comment:
Why has this been commented out? It seems to make sense to me.

Reply:
for consistency, I sticked to blur posts.